### PR TITLE
Added possibility to expose estimated wrenches on ROS topics

### PR DESCRIPTION
**Pull request by francesco-romano**
Even if the branch was originally conceived to be temporary,
a proper solution will take too much time to be coded.

### CODE REVIEW

File: src/devices/wholeBodyDynamics/app/wholebodydynamics-icub-eth-six-fts.xml [hunk 1]
```diff
@@ -8,6 +8,8 @@
         <param name="fixedFrameGravity">(0,0,-9.81)</param>
         <param name="defaultContactFrames">(l_hand,r_hand,root_link,l_sole,r_sole,l_lower_leg,r_lower_leg,l_elbow_1,r_elbow_1)</param>
         <param name="imuFrameName">imu_frame</param>
+        <param name="publishOnROS">true</param>
```
Comment (traversaro) [inline src/devices/wholeBodyDynamics/app/wholebodydynamics-icub-eth-six-fts.xml]: This is working fine even if `roscore` is not running, right?

File: src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 2]
```diff
@@ -630,6 +677,14 @@ bool WholeBodyDynamicsDevice::loadSettingsFromConfig(os::Searchable& config)
         return false;
     }
 
+    yarp::os::Value falseValue;
+    falseValue.fromString("false");
+    if (prop.check("publishOnROS", falseValue, "Checking ROS enabled").asBool()) {
```
Comment (traversaro) [inline src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: Can you add this two options to the device documentation in `WholeBodyDynamicsDevice.h`?

File: src/devices/wholeBodyDynamics/app/wholebodydynamics-icub-eth-six-fts.xml (hunk 1 above)
Comment (francesco-romano) [inline src/devices/wholeBodyDynamics/app/wholebodydynamics-icub-eth-six-fts.xml]: good question. We have to try it.

File: src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp (hunk 2 above)
Comment (francesco-romano) [inline src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: done

**Comment by traversaro** [comment]
Travis is failing for the usual `ALLOW_IDL_GENERATION` for using `yarp_to_idl` etc etc.

**Comment by francesco-romano** [comment]
Do you think I should use the other macro?

**Comment by traversaro** [comment]
It is probably the easiest solution.

**Comment by francesco-romano** [comment]
Ok. I'll do it (next week..)

**Comment by traversaro** [comment]
Opened the issue https://github.com/robotology/codyco-modules/issues/237 to avoid forgetting about the IDL cleanup. I am a bit afraid of merging until we check that everything works fine even without roscore running.

**Comment by traversaro** [comment]
What is the status of this? This was originally used for the CoDyCo Y4/Andy demos for RViz, what you are currently using for this @claudia-lat @Yeshasvitvs @lrapetti @diegoferigo ?

**Comment by diegoferigo** [comment]
Waiting the replies from others, I do not remember to have ever had the need of having the estimated torques in ROS. I don't know what was the original purpose of this feature, probably @francesco-romano remembers.

Right now we use the probabilistic estimator (berdy) included in the framework of human-dynamic-estimation. Probably in this PR (just a guess) we tried to get the output of the deterministic estimator and visualize it in RViz.

**Comment by traversaro** [comment]
Just for clarity: this PR was about plotting in RViz the **estimated contact force/torque**, not the estimated joint torques.

**Comment by traversaro** [comment]
Ok, great. I think that answers it, we can close this PR.

**Pull request closed without merging**